### PR TITLE
fixes: WebTerminal widget should send resize events

### PR DESCRIPTION
**Pull request by sahil143**
fixes: https://issues.redhat.com/browse/ODC-5773

Screenshots: while pasting a command that exceeds 80 cols
Before:
Screenshot 2021-05-17 at 8 04 44 PM
After:
Screenshot 2021-05-17 at 8 04 55 PM

### CODE REVIEW

**Review by JPinkney — APPROVED** [review]
Seems to work well for me! I did notice a couple things:
1. If you continuously type you'll eventually get the horizontal scroll bar at the bottom of the terminal. This happens about 15 characters before the end of the line in the terminal
2. When you re-size from big terminal -> small terminal -> big terminal the words have this weird text overlap thing. Though I'm not sure if we can control that from our side.

Both of those issues should be shown in the gif below. Their fixes (if any) should probably be brought in a different PR. I just wanted to let you know!

openshift-console-test

**Comment by invincibleJai** [comment]
Thanks @sahil143 , have verified the changes and it works as expected with copy and paste of long strings/commands not seeing any wrapping.

i too observe the horizontal scrollbar for a moment as mentioned above and another scenario is on resize cursor position is changed

**Comment by sahil143** [comment]
@invincibleJai @JPinkney I am not able to reproduce the horizontal scrollbar issue.

scrollbar


As for the second issue 

> When you re-size from big terminal -> small terminal -> big terminal the words have this weird text overlap thing. Though I'm not sure if we can control that from our side.

this seems to be a problem with xtermjs. there is already an open issue for this https://github.com/xtermjs/xterm.js/issues/3065

**Comment by invincibleJai** [comment]
/approve

verified the changes on the cluster  and work as expected

File: frontend/packages/console-app/src/components/cloud-shell/Terminal.tsx [hunk 1]
```diff
@@ -24,64 +25,74 @@ export type ImperativeTerminalType = {
...
-    loadAttachAddon: (addOn: ITerminalAddon) => {
-      if (!terminal.current) return;
-      terminal.current.loadAddon(addOn);
-    },
-  }));
+    React.useEffect(() => {
+      const term = terminal.current;
+      term.onData(onData);
+      term.onResize(handleResize);
+    }, [onData, handleResize]);
```
Comment (christianvogt) [inline frontend/packages/console-app/src/components/cloud-shell/Terminal.tsx:68]: Dispose of old listeners.

**Comment by sahil143** [comment]
/hold

failing ci/prow/e2e-gcp-console

**Pull request merged**